### DISCUSSION TITLE
How to handle unexpected errors thrown within the root layout in a Next 14 application?

### DISCUSSION THREAD

**Discussion by karimyilmaz** [vercel/next.js#65196] [Help] [answered]
### Summary

I want to handle unexpected errors in my Next 14 application. My goal is to have a global error fallback UI that will handle all errors thrown within my root layout.

Based on the [documentation](https://nextjs.org/docs/app/api-reference/file-conventions/error#global-errorjs), I could create a global error file inside my app directory which is responsible for handling all unexpected errors.

My project directory looks like the following:

![](https://i.stack.imgur.com/d2OWb.png)

While my global error page component:

![](https://i.stack.imgur.com/VVL5i.png)

For testing purposes, I threw an error inside my About page.

![](https://i.stack.imgur.com/QVSgO.png)

I expected the global error page component to get executed and replace the root layout. Instead, I got the following:

![](https://i.stack.imgur.com/L4CS1.png)

Am I doing something wrong?

**Comment by icyJoseph** [comment]
Is the problem, perhaps, that the About page is in another route group?

**Reply by karimyilmaz** [reply]
No, because I recently tried to throw an error inside the home page I got the same result. The global error page component didn't execute. 
Based on the documentation, the global error will handle all errors thrown inside of the root layout; this means even if the about page is in another route group it is expected for the global error to handle errors thrown inside of the about page as well since the about page eventually is rendering inside of the root layout.

**Reply by icyJoseph** [reply, marked as answer]
Could you just verify one more thing?

In the docs it also says:

> global-error.js is only enabled in production. In development, our error overlay will show instead.

Could you build, your app, and try again?

**Reply by karimyilmaz** [reply]
It worked after building. Thank you.
Do the same applies for error.js?

**Reply by icyJoseph** [reply]
I think, in that case, you'll get the error overlay, and when closing it, you should see the error boundary, IIRC
